Build a regression model.

In [1]:
import pandas as pd
import statsmodels.api as sm

In [2]:
Bike_Stations_Joined = pd.read_csv('Bike_Stations_Joined.csv')

In [3]:
Bike_Stations_Joined

index   latitude  longitude  available_bikes  empty_slots  \
0        0  25.729275 -80.241581                5           11   
1        1  25.732522 -80.233244                7            9   
2        2  25.751100 -80.202641                5            9   
3        3  25.760082 -80.196217                7            9   
4        4  25.761884 -80.193101               10            4   
..     ...        ...        ...              ...          ...   
159    159  25.785199 -80.140549                0           15   
160    160  25.797583 -80.132518                3           13   
161    161  25.776130 -80.194220                6            0   
162    162  25.776620 -80.141910               15            9   
163    163  25.792605 -80.193664                0            0   

                                       name  FS Restaurant Distance  \
0        000605 - Virginia St & Florida Ave                      80   
1      000606 - S Bayshor Dr & Aviation Ave                      80   
2        000612 - SE Brickell Ave & 25th Rd                     238   
3                 000616 - 150 SW 15th Road                     239   
4        000617 - S Miami Ave & 13th Street                      44   
..                                      ...                     ...   
159                   000145 - 14th & Alton                     121   
160     000199 - Washington Ave & Dade Blvd                     196   
161               000674 - 48 NW 2nd Street                     116   
162  000130 - West Ave 6-7 St (Canopy Park)                     233   
163     000002 - DEMO - Deco Bike Warehouse                     114   

     Station Distance  yelp_index  yelp_rating  yelp_distance  Bike_Usage  
0                   0           0          5.0     173.754030       68.75  
1                   0           1          5.0     540.432571       56.25  
2                   0           2          4.0     305.416309       64.29  
3                   0           3          5.0     109.908894       56.25  
4                   0           4          5.0     133.216789       28.57  
..                ...         ...          ...            ...         ...  
159                 0         159          5.0     222.801721      100.00  
160                 0         160          5.0      20.431553       81.25  
161               181         161          5.0      28.004793        0.00  
162                 0         162          4.5     317.559921       37.50  
163                 0         163          5.0     371.815630        0.00  

[164 rows x 12 columns]

In [8]:
relevant_metrics = Bike_Stations_Joined[['Bike_Usage', 'FS Restaurant Distance', 'Station Distance', 'yelp_distance','available_bikes','empty_slots']].copy()

In [9]:
relevant_metrics['total slots']=relevant_metrics['empty_slots']+relevant_metrics['available_bikes']

In [12]:
relevant_metrics[['total slots','FS Restaurant Distance', 'Station Distance', 'yelp_distance']].corr()

total slots  FS Restaurant Distance  Station Distance  \
total slots                1.000000                0.027478         -0.070341   
FS Restaurant Distance     0.027478                1.000000         -0.091677   
Station Distance          -0.070341               -0.091677          1.000000   
yelp_distance              0.038370                0.437010         -0.041163   

                        yelp_distance  
total slots                  0.038370  
FS Restaurant Distance       0.437010  
Station Distance            -0.041163  
yelp_distance                1.000000

In [13]:
y = relevant_metrics['total slots']
X = relevant_metrics[['FS Restaurant Distance', 'Station Distance', 'yelp_distance']]
X = sm.add_constant(X)
X.head()

const  FS Restaurant Distance  Station Distance  yelp_distance
0    1.0                      80                 0     173.754030
1    1.0                      80                 0     540.432571
2    1.0                     238                 0     305.416309
3    1.0                     239                 0     109.908894
4    1.0                      44                 0     133.216789

In [15]:
model = sm.OLS(y, X) #instantiate
results = model.fit() #fit the model (this is where OLS is actually being run)


Provide model output and an interpretation of the results. 

In [16]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:            total slots   R-squared:                       0.006
Model:                            OLS   Adj. R-squared:                 -0.012
Method:                 Least Squares   F-statistic:                    0.3353
Date:                Sun, 19 Nov 2023   Prob (F-statistic):              0.800
Time:                        13:27:13   Log-Likelihood:                -485.82
No. Observations:                 164   AIC:                             979.6
Df Residuals:                     160   BIC:                             992.0
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                     15

There is no real significant correlation between the distance variables in being able to predict the number of total bikes at a nearby bikestation. 
The null hypothesis of no correlation can be accepted because all three variables have p values greater than 0.05. 
This matches up with the coeff values all being relatively low (0.001 or lower). 
Overall this is a weak model in being able to predict total bikes as the adj r-squared is essentially showing that it can't predict even 1% of values (-0.012).

# Stretch

How can you turn the regression model into a classification model?